delta/sdk/src/test/scala/ch/epfl/bluebrain/nexus/delta/sdk/resources/ResourcesImplSpec.scala

```diff
@@ -600,6 +600,59 @@ class ResourcesImplSpec
 
     }
 
+    "undeprecating a resource" should {
+
+      "succeed" in {
+        givenADeprecatedResource { id =>
+          resources.undeprecate(id, projectRef, Some(schemas.resources), 2).accepted.deprecated shouldEqual false
+          resources.fetch(id, projectRef, None).accepted.deprecated shouldEqual false
+        }
+      }
+
+      "reject if resource does not exist" in {
+        givenADeprecatedResource { _ =>
+          val wrongId = nxv + genString()
+          resources
+            .undeprecate(wrongId, projectRef, None, 2)
+            .assertRejected[ResourceNotFound]
+        }
+      }
+
+      "reject if resource is revision is incorrect" in {
+        givenADeprecatedResource { id =>
+          resources
+            .undeprecate(id, projectRef, None, 4)
+            .assertRejected[IncorrectRev]
```

Comment:
Would it be overkill to check the resource remains undeprecated?

Reply:
Added where appropriate